crates/levm/src/call_frame.rs

```diff
@@ -32,4 +32,26 @@ impl CallFrame {
     pub fn pc(&self) -> usize {
         self.pc
     }
+
+    pub fn jump(&mut self, jump_address: U256) {
+        if !self.valid_jump(jump_address) {
+            // probably should halt/panic
+            dbg!("Invalid jump");
+            return;
+        }
+        self.pc = jump_address.as_usize() + 1;
+    }
+
+    fn valid_jump(&self, offset: U256) -> bool {
+        // In the future this should be the Opcode::Invalid and halt
+        self.opcode_at(offset).eq(&Opcode::JUMPDEST)
+    }
+
+    fn opcode_at(&self, offset: U256) -> Opcode {
+        self.bytecode
+            .get(offset.as_usize())
+            .copied()
+            .map(Opcode::from)
+            .unwrap_or(Opcode::STOP)
+    }
```

Comment:
Firstly, I misunderstood the purpose of the `unwrap_or`, I thought that we could abstract that logic into `opcode_at` to reuse it in other functions if needed. After discussing this, I came up with this new suggestion:

```suggestion
    fn valid_jump(&self, offset: U256, bytecode: &Bytes) -> bool {
        // In the future this should be the Opcode::Invalid and halt
        bytecode
            .get(offset.as_usize())
            .copied()
            .map(Opcode::from)
            .map(|opcode| opcode.eq(&Opcode::JUMPDEST))
            .is_some_and(|is_jumpdest| is_jumpdest)
    }
```

Reply:
We can later define an `opcode_at` function used by `next_opcode` and `valid_jump` that returns an `Option<Opcode>`.

Reply:
Just updated it and added the `opcode_at` function, changing `next_opcode` and `valid_jump` to use it